### ISSUE TITLE
Setup issues

### ISSUE THREAD

**Issue by BrendanxP** [erik73/addon-mail#50]
# Problem/Motivation

> I cannot get the setup right, have struggled a lot!

## Expected behavior

>Being able to send and receive emails using my domain.

## Actual behavior

> Nothing

## Steps to reproduce

> I have my own domain using Cloudflare DNS connect using a reverse proxy from Nginx Proxy Manager HassOS add-on to secure the connection.  Though, locally I can still use 192.168.178.100:8123 over http (no s) and send tts messages to my google devices. Before I got my own domain abc***.de, I used abc***.duckdns.org, which I also linked through the proxy. So the old duckdns url still works encrypted and all, but I do not really use it anymore.

I am getting stuck at setting up the email client because of the Nginx reverse proxy. I have no idea what links I need to connect with what ports and which records I need to make in Cloudflare. Nginx Proxy Manger also only allows for http or https types, I was not really sure if this would be an issue. Additionally, my router cannot forwards port 25 as internal nor external address. Though, I did forward 465 to 192:168.178.100:465 for tcp, 587 to 192:168.178.100:587 for tcp and 993 to 192:168.178.100:993 for tcp.

Using a combination of tutorials and bits of information online I made the following records, proxy settings and add-on configurations (these include a couple for the home assistant to domain itself)
Domain = bc***.de
Host N. = Hostname given on info page of mailserver add-on.
Public IP = my public IP where the raspberri pi with home assistant is located.
Cloudflare IP = some IP I got when doing nslookup on my domain name.
Random IP = the Cloudflare IP + 1, as some tutorials stated I should do??

![cloudflare-records](https://user-images.githubusercontent.com/25636915/108776832-36691300-7563-11eb-895f-457196c045b7.png)

![nginx-proxy](https://user-images.githubusercontent.com/25636915/108776927-56003b80-7563-11eb-8a2b-c8d593527c62.jpg)

![configuration](https://user-images.githubusercontent.com/25636915/108778924-6d8cf380-7566-11eb-8814-15474c868898.png)

Obviously, the log gives a whole lot of:
`mail.info: Feb 22 22:13:29 postfix/smtp[813]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Operation timed out
mail.info: Feb 22 22:13:29 postfix/smtp[810]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Operation timed out
mail.info: Feb 22 22:13:29 postfix/smtp[812]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Operation timed out
mail.info: Feb 22 22:13:29 postfix/smtp[811]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Operation timed out
mail.info: Feb 22 22:13:29 postfix/smtp[814]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Operation timed out
mail.info: Feb 22 22:13:59 postfix/smtp[813]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.27]:25: Operation timed out
mail.info: Feb 22 22:13:59 postfix/smtp[810]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.27]:25: Operation timed out
mail.info: Feb 22 22:13:59 postfix/smtp[812]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.27]:25: Operation timed out
mail.info: Feb 22 22:13:59 postfix/smtp[811]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.27]:25: Operation timed out
mail.info: Feb 22 22:13:59 postfix/smtp[814]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.27]:25: Operation timed out
mail.info: Feb 22 22:14:29 postfix/smtp[811]: connect to alt2.gmail-smtp-in.l.google.com[142.250.4.27]:25: Operation timed out
mail.info: Feb 22 22:14:29 postfix/smtp[812]: connect to alt2.gmail-smtp-in.l.google.com[142.250.4.27]:25: Operation timed out
mail.info: Feb 22 22:14:29 postfix/smtp[814]: connect to alt2.gmail-smtp-in.l.google.com[142.250.4.27]:25: Operation timed out
mail.info: Feb 22 22:14:29 postfix/smtp[813]: connect to alt2.gmail-smtp-in.l.google.com[142.250.4.27]:25: Operation timed out
mail.info: Feb 22 22:14:29 postfix/smtp[810]: connect to alt2.gmail-smtp-in.l.google.com[142.250.4.27]:25: Operation timed out`

The log starts as follow:

 ` [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 

 Add-on: Mailserver
 Postfix and Dovecot mail server for Home Assistant, with Postfix Admin web interface

 Add-on version: 1.1.4
 You are running the latest version of this add-on.
 System: Home Assistant OS 5.10  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2021.1.5
 Home Assistant Supervisor: 2021.02.11

 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.

[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] 10-create-config.sh: executing... 
[23:36:09] INFO: Let's Encrypt certs will be used...
<p>Database is up to date: 1843/0 </p>
[cont-init.d] 10-create-config.sh: exited 0.
[cont-init.d] ~-socklog: executing... 
[cont-init.d] ~-socklog: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[23:36:11] INFO: Starting Postfix....
[23:36:11] INFO: Starting PHP-FPM...
listening on /run/systemd/journal/dev-log, gid=65534, uid=65534, starting.
mail.info: Feb 22 22:36:12 postfix/master[652]: daemon started -- version 3.5.9, configuration /etc/postfix
[23:36:12] INFO: Starting Dovecot....
mail.info: Feb 22 22:36:12 dovecot: master: Dovecot v2.3.13 (89f716dc2) starting up for imap, lmtp
[23:36:13] INFO: Starting NGinx.... `

**Comment by erik73** [comment]
Hi.

First of all, the Nginx Proxy Manager should not have anything to do with this. Remove the redirection host.

You already have an A-record in dns. The line with: "A" "domain name" "public ip". Keep that, and remove the A-record for the mail server.

The MX record should be the same. Just your.domain.com (or whatever your domain name is).

The hostname in the add-on configuration should also be changed to your.domain.com. This corresponds to your A-record.

So in short. You should have an A-record for your.domain.com (as you already have) and a MX record for your.domain.com.

To send email, it is most likely you need to use a smtp relay host. Most providers have one. You will probably not be able to send emails directly to the internet. That I why google is complaining in the logs. Your server is not trusted.

**Comment by erik73** [comment]
And I forgot. Port 25 is required to receive emails. If you cannot forward port 25 from your public IP to your Home Assistant local IP, I think you are in trouble...

**Comment by BrendanxP** [comment]
Thanks for your message @erik73 ! It clears up a lot of things. 

As my ISP is blocking forwarding of some ports, including 25, it is currently impossible for me to get this server working. I would have to put my modem into bridge mode but that's no option in my current hardware living situation, sadly. 

I still do not understand two things;
1. The given hostname on the info page, for me: "3****a-mailserver", where would this value be used in a completed setup? As it is not used in the record nor the email address, right?
2. In the add-on config, hostname and domain are equal now. Why is there a distinction made between the two, when would they not be equal?

**Add-on config:**
hostname: abc.de
domain: abc.de

**DNS Records:**
Type; Name; Content
A; abc.de; 9***Public IP***4
MX; abc.de; abc.de

**Comment by erik73** [comment]
Hi.

Regarding 1: The hostname you see on the info page is the internal DNS name for the addon itself. All addons have hostnames based on the addon slug. It is generated automatically by the supervisor, and you don´t need to worry about it.
The hostname you enter in the addon config is used to setup Postfix. Postfix will in your case present itself as abc.de for all external connections. 

2. You can, if you want, have a hostname like "mail.abc.de". That has to be an A-record for that in DNS. Then you have a MX record that points to the host mail.abc.de for the abc.de domain.
Since you already have an A-record for abc.de it does not make sense to add another hostname.
In a corporate environment it makes sense. Then you might have A-records mail1.abc.de mail2.abc,de mail3.abc.de etc. They all have different IP-addresses, and they all listen on the SMTP ports. The corresponding MX record would then point to all 3 servers, and have different priorities.

**Comment by BrendanxP** [comment]
Hey @erik73,

That makes a lot of sense now! Thanks a lot for your time and knowledge. When I move out of this rental apartment I'll give this mail server another shot for sure!

Goodluck with continuing this project :)
